### PR TITLE
log cmd info for regresstion test

### PR DESCRIPTION
**Pull request by RunningLeon**
## Motivation
log cmd to log for regresstion test
Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

update `tools/regresstion_test.py` to include cmd info to log for both running `deploy.py` and `test.py`

There are totally three output log files:
- `convert.log` is the log file of running `tools/deploy.py`
- `backend_test.log` is the log file of running `tools/test.py` with backend model file.
- `sdk_test.log` is the log file of running `tools/test.py` with sdk + backend model file.

### example logging info 
```
2022-09-30 08:55:08,705 - mmdeploy - INFO - ----------------------------------------------------------------------------------------------------
2022-09-30 08:55:08,705 - mmdeploy - INFO - Start running cmd
python3 ./tools/deploy.py \
/root/workspace/mmdeploy/configs/mmseg/segmentation_tensorrt-fp16_static-1024x2048.py \
/root/workspace/mmsegmentation/configs/fastscnn/fast_scnn_lr0.12_8x4_160k_cityscapes.py \
"/root/workspace/mmdeploy_checkpoints/mmseg/fastscnn/fast_scnn_lr0.12_8x4_160k_cityscapes_20210630_164853-0cec9937.pth" \
"../mmsegmentation/tests/data/pseudo_cityscapes_dataset/leftImg8bit/frankfurt_000000_000294_leftImg8bit.png" \
--work-dir "../mmdeploy_regression_working_dir/mmseg/fastscnn/tensorrt/static/fp16/fast_scnn_lr0.12_8x4_160k_cityscapes_20210630_164853-0cec9937" \
--device cuda:0  \
--log-level INFO \
--dump-info \
--test-img ../mmsegmentation/tests/data/pseudo_loveda_dataset/img_dir/0.png

2022-09-30 08:55:08,705 - mmdeploy - INFO - Logging log to 
../mmdeploy_regression_working_dir/mmseg/fastscnn/tensorrt/static/fp16/fast_scnn_lr0.12_8x4_160k_cityscapes_20210630_164853-0cec9937/convert.log
```

### example of output convert log file
```
Command:
python3 ./tools/deploy.py \
/root/workspace/mmdeploy/configs/mmseg/segmentation_tensorrt-fp16_static-1024x2048.py \
/root/workspace/mmsegmentation/configs/fastscnn/fast_scnn_lr0.12_8x4_160k_cityscapes.py \
"/root/workspace/mmdeploy_checkpoints/mmseg/fastscnn/fast_scnn_lr0.12_8x4_160k_cityscapes_20210630_164853-0cec9937.pth" \
"../mmsegmentation/tests/data/pseudo_cityscapes_dataset/leftImg8bit/frankfurt_000000_000294_leftImg8bit.png" \
--work-dir "../mmdeploy_regression_working_dir/mmseg/fastscnn/tensorrt/static/fp16/fast_scnn_lr0.12_8x4_160k_cityscapes_20210630_164853-0cec9937" \
--device cuda:0  \
--log-level INFO \
--dump-info \
--test-img ../mmsegmentation/tests/data/pseudo_loveda_dataset/img_dir/0.png
.......
2022-09-30 08:57:03,178 - mmdeploy - INFO - visualize pytorch model success.
2022-09-30 08:57:03,178 - mmdeploy - INFO - All process success.
```

## BC-breaking (Optional)

None

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: tools/regression_test.py [hunk 1]
```diff
@@ -519,6 +520,52 @@ def get_info_from_log_file(info_type: str, log_path: Path,
...
+    """
+    return_code = 1
+    import platform
+    system = platform.system().lower()
+
+    if system == 'windows':
+        sep = r'`'
+    else:  # 'Linux', 'Darwin'
+        sep = '\\'
+    cmd_str = f' {sep}\n'.join(cmd_lines) + '\n'
```
Comment (irexyc) [inline tools/regression_test.py:540]: use` cmd_str = f' '.join(cmd_lines) on windows`

Comment (RunningLeon) [reply tools/regression_test.py:540]: > use` cmd_str = f' '.join(cmd_lines) on windows`
@irexyc 
It's for better human visualization of multiple lines. Is the separator ` OK for windows?

**Review by irexyc — APPROVED** [review]

File: tools/regression_test.py [hunk 2]
```diff
@@ -519,6 +520,53 @@ def get_info_from_log_file(info_type: str, log_path: Path,
...
+
+    if system == 'windows':
+        sep = r'`'
+    else:  # 'Linux', 'Darwin'
+        sep = '\\'
+    cmd_for_run = ' '.join(cmd_lines)
+    cmd_for_log = f' {sep}\n'.join(cmd_lines) + '\n'
+    parent_path = log_path.parent
+    if not parent_path.exists():
+        parent_path.mkdir(parents=True, exist_ok=True)
```
Comment (AllentDan) [inline tools/regression_test.py:543]: why not using `mmcv.mkdir_or_exist`?

**Review by AllentDan — APPROVED** [review]

File: tools/regression_test.py (hunk 2 above)
Comment (RunningLeon) [reply tools/regression_test.py:543]: Because it's the instance of `Path` class, not of `str` type.

Comment (AllentDan) [reply tools/regression_test.py:543]: but it should contains the string. Anyway, codes can be kept.

File: tests/regression/mmseg.yml [hunk 3]
```diff
@@ -211,8 +211,8 @@ models:
     model_configs:
       - configs/apcnet/apcnet_r50-d8_512x1024_40k_cityscapes.py
     pipelines:
-      - *pipeline_ort_dynamic_fp32
-      - *pipeline_trt_dynamic_fp16
+      - *pipeline_ort_static_fp32
```
Comment (lvhan028) [inline tests/regression/mmseg.yml:214]: what's the motivation of make changes to `mmseg.yml`?

Comment (RunningLeon) [reply tests/regression/mmseg.yml:214]: apcnet only supports static shape as described https://github.com/open-mmlab/mmdeploy/issues/1139

**Pull request merged**